### ISSUE TITLE
Only run travis push builds on trunk

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8767]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#8767|
|Type|enhancement|
|Created|2016-08-18 15:38:45Z|
|Branch|https://github.com/twisted/twisted/tree/8767-travis-trunk-push|

<details><summary>Searchable metadata</summary>

```
trac-id__8767 8767
type__enhancement enhancement
reporter__adiroiban adiroiban
priority__normal normal
milestone__None None
branch__8767_travis_trunk_push 8767-travis-trunk-push
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1471534725756361 1471534725756361
changetime__1471600411138606 1471600411138606
version__None None
owner__Amber_Brown__hawkowl_____ Amber Brown <hawkowl@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_1"></a>|@adiroiban removed owner|
|-|-|

PR at https://github.com/twisted/twisted/pull/506

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban commented|
|-|-|

btw... appveyor has an ugly `skip_branch_with_pr: true` option ... and for branches created on the main repo, you will push the branch before the PR... so appveyor will start  to run the tests ... and then when you create the PR for the first commit you will see  that both branch and pr tests are running on appveyor

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

LGTM

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set owner to @hawkowl<br>@hawkowl set status to `closed`|
|-|-|

In changeset 0433e91fa79d9a570b404c2d80d652167246e589
```
#!CommitTicketReference repository="" revision="0433e91fa79d9a570b404c2d80d652167246e589"
Merge 8767-travis-trunk-push: Make travis-ci only run on PRs, trunk, and release branches

Author: adiroiban
Reviewer: hawkowl
Fixes: #8767
```
